### DISCUSSION TITLE
Why is TestContext not exposed as part of the api?

### DISCUSSION THREAD

**Discussion by isaacgr** [nodejs/help#4767] [Q&A]
I'd like to use it for typing as I want to pass it around to some generic functions. 

It also seems like it would be useful to allow extending it to build a custom test runner from this native solution. This is really what I'm trying to achieve but im not sure how to do it without the TestContext being available externally.

**Comment by cjihrig** [comment]
> Why is TestContext not exposed as part of the api?

The reason the `TestContext` constructor was not exposed publicly is because its only option is a private `Test` object, which exposes the internals of the test runner. The idea originally was to reduce the ways that users could tamper with the test runner's internals.

> It also seems like it would be useful to allow extending it

If we exposed it in its current state, you wouldn't be able to do much with it. All of its methods operate on its private `Test` object, which you wouldn't have access to outside of where the class is defined.

If people want to open up all of the internals so that the test runner can be extended everywhere, that's a separate discussion. My personal opinion is that we shouldn't do that. I would like to see some extension points though. For example, the first thing that comes to mind is an API for binding an arbitrary assertion library to the test runner in order to implement `plan()`.
